src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/IndexStateManagementIndices.kt

```diff
@@ -40,15 +41,15 @@ class IndexStateManagementIndices(
 
     private val logger = LogManager.getLogger(javaClass)
 
-    val indexStateManagementMappings = javaClass.classLoader.getResource("mappings/opendistro-ism-config.json").readText()
-
-    val indexStateManagementHistoryMappings = javaClass.classLoader.getResource("mappings/opendistro-ism-history.json").readText()
-
     fun initIndexStateManagementIndex(actionListener: ActionListener<CreateIndexResponse>) {
         if (!indexStateManagementIndexExists()) {
             val indexRequest = CreateIndexRequest(INDEX_STATE_MANAGEMENT_INDEX)
-                    .mapping(_DOC, indexStateManagementMappings, XContentType.JSON)
+                    .mapping(_DOC, indexStateManagementMappings(), XContentType.JSON)
             client.create(indexRequest, actionListener)
+        } else {
+            IndexUtils.updateIndexMapping(INDEX_STATE_MANAGEMENT_INDEX,
```

Comment:
Curious what behavior is when this updateIndexMapping is called twice in a row from different nodes/requests?

Reply:
Then the mapping will be updated twice. Unless user is using different ISM plugin version it would just be updating the mapping to same version twice.